nrel/hive/dispatcher/instruction_generator/instruction_function.py

```diff
@@ -0,0 +1,45 @@
+from __future__ import annotations
+
+from typing import Tuple, Callable, TYPE_CHECKING, Union
+
+from nrel.hive.dispatcher.instruction_generator.instruction_generator import InstructionGenerator
+from nrel.hive.state.simulation_state.simulation_state import SimulationState
+from nrel.hive.runner.environment import Environment
+from nrel.hive.dispatcher.instruction.instruction import Instruction
+
+
+InstructionFunction = Callable[[SimulationState, Environment], Tuple[Instruction, ...]]
+
+
+class AnonGenerator(InstructionGenerator):
+    """
+    A class that wraps an instruction function as an instruction generator
+    """
+
+    def __init__(self, instruction_function: InstructionFunction):
+        self.instruction_function = instruction_function
+
+    @property
+    def name(self) -> str:
```

Comment:
i _think_ you set up instruction generators to be stored in a Map[str, InstructionGenerator] where the key comes from this name. if that's the case, this design fails if there are more than one anonymous InstructionGenerators added to the system.

two ideas:
- while looping through `instruction_generator_from_function` below, enumerate the iterable and append the index number when creating the name in the __init__ of AnonGenerator
- make users responsible for providing a name, which is probably the right move, though it means they have to pass along a Tuple[str, Callable[etc]] instead of just the callable

Reply:
aha yes, good catch; how about if we use the incoming function name?

Reply:
that's going to be something generated by the interpreter like "lambda$a2354h8-aggahds" or something like that, right? is that helpful for the user? again, if there is a runtime error in their code, it might not be clear which of multiple IGs is the one that's responsible.

Reply:
it's a good start though. maybe i should back off and let that through and we worry about fixing it if people have issues later.

Reply:
it's a good start though. maybe i should back off and let that through and we worry about fixing it if people have issues later.

Reply:
that looks good enough to me, as it should likely correspond to something the user can make sense of. thanks for confirming!